earthfile2llb/converter.go

```diff
@@ -1016,17 +1016,17 @@ func (c *Converter) Healthcheck(ctx context.Context, isNone bool, cmdArgs []stri
 }
 
 // Import applies the IMPORT command.
-func (c *Converter) Import(ctx context.Context, importStr, as string, isGlobal bool) error {
+func (c *Converter) Import(ctx context.Context, importStr, as string, isGlobal, currentlyPrivileged, allowPrivilegedFlag bool) error {
 	err := c.checkAllowed("IMPORT")
 	if err != nil {
 		return err
 	}
-	return c.varCollection.Imports().Add(importStr, as, isGlobal)
+	return c.varCollection.Imports().Add(importStr, as, isGlobal, currentlyPrivileged, allowPrivilegedFlag)
 }
 
-// ResolveReference resolves a reference's build context given the current state: relativity to the Earthfile, imports etc.
+// ResolveReference resolves a reference's build context given the current state: relativity to the Earthfile, imports etc. NOTE: this function is used to support UDFs (user defined functions), and does not make use of the allowPrivileged values returned by Deref.
 func (c *Converter) ResolveReference(ctx context.Context, ref domain.Reference) (*buildcontext.Data, error) {
-	derefed, err := c.varCollection.Imports().Deref(ref)
+	derefed, _, _, err := c.varCollection.Imports().Deref(ref)
```

Comment:
Wouldn't this information be needed for UDCs too?

Reply:
I was on the fence about this one, because even without it, it could be abused:

```
# github.com/scary/repo
UNSAFE:
    COMMAND
    RUN rm -rf /
    
# local Earthfile
foo:
    LOCALLY
    DO github.com/scary/repo+UNSAFE
```

since the unsafe command doesn't require priviledged mode, it could be pulled into a locally target and could do a fair amount of damage; I was worried that adding a `--allow-priviledged` flag could give users an unsafe reassurance that it would protect them in this case. If we did extend it for this, we could have it simply block `RUN --priviledged` calls, or calls to other LOCALLY targets.

maybe it's still good to extend the proection here for users who want to use UDCs within a containerized target?

Reply:
I would do it still, because even in your example, the user had to specify LOCALLY to get into trouble first

Reply:
I just submitted a follow up commit with an implementation for this.